File: modules/android/examples/GetStartedJava/app/src/main/java/com/couchbase/android/getstarted/java/MainActivity.java
package com.couchbase.android.getstarted.java;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;


public class MainActivity extends AppCompatActivity {
     private MainViewModel model;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        model = new ViewModelProvider(this).get(MainViewModel.class);

        TextView statusView = findViewById(R.id.replState);
        Button runButton = findViewById(R.id.runIt);

        // Extremely simple: allows starting one replicator on top of another...
        runButton.setOnClickListener(v -> model.runIt().observe(this, statusView::setText));
    }

    @Override
    protected void onStop() {
        super.onStop();
        model.stopIt();
    }
}

File: modules/android/examples/GetStartedKotlin/app/src/main/java/com/couchbase/android/getstarted/kotlin/MainActivity.kt
package com.couchbase.android.getstarted.kotlin

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import org.koin.androidx.viewmodel.ext.android.viewModel


class MainActivity : AppCompatActivity() {
    private val model by viewModel<MainViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val statusView = findViewById<TextView>(R.id.replState)
        val runButton = findViewById<Button>(R.id.runIt)

        // Extremely simple: allows starting one replicator on top of another...
        runButton.setOnClickListener {
            model.runIt().observe(this) { statusView.text = it }
        }
    }
}
